### ISSUE TITLE
warning popup after start of ST3

### ISSUE THREAD

**Issue by JimAmuro** [DamnWidget/anaconda#514]
```
anacondaST3: ERROR - <Anaconda.anaconda_lib.workers.local_worker.LocalWorker object at 0x0000022AE8CEA630> initial check failed because:

connection to localhost:1552 timed out after 0.05s

check that there is Python process executing the anaconda jsonserver.py script running in your system. If there is, check that you can connect to your localhost writing the following script in your Sublime Text 3 console:

import socket; socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect(("localhost", 1552))
```
After 2.0.x update, this warning always popups for the first or second time I start ST3 after I start my computer. I have no idea how to deal with it...

OS: Win 10.0.14393
ST3 3114

**Comment by DamnWidget** [comment]
Anaconda works fine after that? If so, don't worry too much, your OS X is just a bit slow

**Comment by MrLear** [comment]
I have the same issue. I get the same error at startup. Is there a way to fix it?

**Comment by JimAmuro** [comment]
@DamnWidget Yeah it works fine afterwards. It is just annoying...
And my system is Windows 10, not OS X...

**Comment by alejoar** [comment]
Same problem here with Ubuntu 14.04.

**Comment by DamnWidget** [comment]
Is weird that you Operating Systems can't connect to your localhost in less than 0.05s (is much more weirder for @alejoar as for him anaconda is using an Unix Socket Domain).

Can I know your hardware specs please?

**Comment by alejoar** [comment]
```
$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    2
Core(s) per socket:    2
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 69
Model name:            Intel(R) Core(TM) i5-4210U CPU @ 1.70GHz
Stepping:              1
CPU MHz:               800.906
CPU max MHz:           2700,0000
CPU min MHz:           800,0000
BogoMIPS:              4789.15
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              3072K
NUMA node0 CPU(s):     0-3
```

**Comment by MrLear** [comment]
I upgraded the package to v2.0.3. It didn't resolve the issue for me. I still get the same error.

**Comment by JimAmuro** [comment]
Same here. Didn't resolve the problem.
Processor: Intel(R) Core(TM) i7-4700MQ CPU @ 2.40GHz, 2401 Mhz, 4 Core(s), 8 Logical Processor(s)
I don't think it is a slow CPU...

**Comment by DamnWidget** [comment]
Maybe slow disks or too much IO in your device, anyway, is not normal that your OS is unable to open a connection to the loopback interface in less than 0.1 seconds

**Comment by alejoar** [comment]
@DamnWidget I didn't encounter the problem today, seems to be fixed for me.

**Comment by DamnWidget** [comment]
Glad to read @alejoar

**Comment by webtotma** [comment]
Similar problem on windows 10 x64, lubuntu x86 15.10.

**Comment by cuiluscnu** [comment]
I use win10 and have the same problem, i change the 

C:\Users\<name>\AppData\Roaming\Sublime Text 3\Packages\Anaconda\anaconda_lib\workers\local_worker.py

line 34:
while not self._status(0.05):

to
while not self._status(0.20):

It work fine, you can try it.

**Comment by timleslie** [comment]
I also see this problem on a freshly installed Windows 7 with the latest version of anaconda. Bumping up the timeout in local_worker.py did not fix the problem for me.

**Comment by narusFR** [comment]
Hello,

I have the same error since 2.0.x too. It happens more often when I open python project directory with multiples .py files inside.

I have to specify that if I remove the following options from my user config, I don't have the error anymore :
	"hot_exit": false,
	"remember_open_files": false,

But the thing is, I want these options. 
Do you guys have these options or similar ones (that affect how ST starts) ?
I think it could be useful for DEVS in order to reproduce the issue and for debug purpose.

Regards.

**Comment by DamnWidget** [comment]
I will redesign how the plugin fires it's jsonserver worker and how it connects to it so you don't get annoyed with the popup when you start ST3 or open a new window containing Python files.

Said that, take into account that is quite weird that you get this warning due slowness in your system, I did test this hundred of times during development in Windows, OS X and Linux and I never got this warning popup just because my system was not fast enough as to stablish a connection with the local worker in less than 0.05s.

For now, just wait until I release a new version with a new approach, seems like v2.1 is coming faster than I expected :(

**Comment by timleslie** [comment]
The fix appears to have solved the problem for me. Thanks for the fast response!

**Comment by JimAmuro** [comment]
The new version works! Thank you for fixing this!

**Comment by Bladefidz** [comment]
For me, who using Windows 10. have newest version, but that error still happening. Uhm, anybody same with me?

**Comment by narusFR** [comment]
@Bladefidz

I do, but I suppose it's our HIPS at work that blocks the connection. It's working fine except this popup, so I changed an option to hide the alerts at anaconda startup.

**Comment by espositofulvio** [comment]
I used to experience this issue on Windows 10/Sublime Text 3 as well, it simply went away after I switched to an SSD replacing my old HDD. Could it be that the js server takes too much time to fire up and thus anaconda can't connect?

**Comment by DamnWidget** [comment]
That could be a reason @espositofulvio I never had these kind of warning but all my boxes use SSD

**Comment by narusFR** [comment]
Also I don't have a SSD at work. I do have a SSD at home where and I don't get the error.

**Comment by JPCoding** [comment]
I experience this issue on Windows 10 and Sublime Text 3 with a SSD as well. If i try to connect to the socket manually with `import socket; socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect(("localhost", 54133))`, i receive this: 

`socket.error: [Errno 10061] No connection could be made because the target machine actively refused it`.

**Comment by DamnWidget** [comment]
Probably a firewall issue @JPCoding ?

**Comment by JPCoding** [comment]
Tested this. Error still remains.

**Comment by DamnWidget** [comment]
What did you test? If the problem is not a firewall then probably there is no JsonServer started for anaconda to connect to, check if there are any `python` processes running in the system using `tasklist` in a CMD.exe or PowerShell

**Comment by JPCoding** [comment]
I disabled the firewall to test if that causes the issue, but error remains. 
There is one `python` process running.

**Comment by DamnWidget** [comment]
Take a look at ` %APPDATA\Anaconda\Logs`

Note that majority of people that gets this warning, have no problems at all after close the popup message, so make sure that anaconda is not working for you after close that message (because probably it is working) and if it works, just ignore the message.

**Comment by JPCoding** [comment]
Actually, there is no `%APPDATA\Anaconda\Logs` folder and I receive continuously the same error message  in the Sublime Text console: 

`anacondaST3: ERROR - <Anaconda.anaconda_lib.workers.local_worker.LocalWorker object at` `0x000002503270D828> initial check failed because:`

`connection to localhost:51936 timed out after 0.2s. tried to connect 7 times during 2 seconds`

`check that there is Python process executing the anaconda jsonserver.py script running in your system. If there is, check that you can connect to your localhost writing the following script in your Sublime Text 3 console:`

`import socket; socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect(("localhost", 51936))`

`If anaconda works just fine after you received this error and the command above worked you can make anaconda to do not show you this error anymore setting the 'swallow_startup_errors' to 'true' in your configuration file.`

Followed by ` ...last message repeated one more time in the last 10s`.

So, no. Anaconda is not working properly, i guess. But maybe i'm missing something, i'm not expert with this. I appreciate your time spending here.

**Comment by DamnWidget** [comment]
What is your `python_interpreter` configuration?

**Comment by JPCoding** [comment]
`python_interpreter` is set to default, so Anaconda should find my interpreter in the `PATH`.

**Comment by DamnWidget** [comment]
Can you run the following in the ST3 console please? (line per line or it will not work)
```python
import subprocess
print(subprocess.getoutput('python -c "import sys; print(sys.version_info);"'))
```
Thanks.

**Comment by JPCoding** [comment]
I get`AttributeError: 'module' object has no attribute 'getoutput'`.

**Comment by DamnWidget** [comment]
Looks like your Python installation is broken. Can you please remove your Python installation and install it again from scratch?

**Comment by JPCoding** [comment]
Sublime Text Build 3126.

**Comment by DamnWidget** [comment]
Looks like the embedded Python interpreter in your ST3 is screwed up... Can you please remove ST3 completely from your machine and reinstall it?

**Comment by JPCoding** [comment]
I did. With a fresh Sublime and the Anaconda package installed i still get the error.

**Comment by DamnWidget** [comment]
If when you try to use `subprocess.getoutput` in your ST3 console you get the error you pasted [here](https://github.com/DamnWidget/anaconda/issues/514#issuecomment-262584387) that means ST3 python interpreter is not working as intended, sadly I can not offer you support for Sublime Text 3 itself beyond what I already did.

Anaconda uses the `subprocess` module from your ST3 python embedded interpreter, you have to understand that I can not possibly fix a problem in your ST3 or fix a problem in ST3 directly in anaconda.

Try to run the commands that I pasted earlier again, if you continue getting that kind of error about `getoutput` not being defined, consider to open a ticket or start a new thread in the official Sublime Text forum.

**Comment by JPCoding** [comment]
Actually, if i enter now the command, i get this `sys.version_info(major=2, minor=7, micro=12, releaselevel='final', serial=0)`.

**Comment by DamnWidget** [comment]
Sorry for the delay @JPCoding but is actually hard to follow up a closed ticket, can we continue this on Gitter or Email?

**Comment by JPCoding** [comment]
No problem. I could resolve my issues. Actually there was no error with anaconda at all, not directly. There was an encoding error during execution, because my path contains special characters. This let the jsonserver not start.

**Comment by DamnWidget** [comment]
Oh, I see, glad to hear :)

**Comment by mch027** [comment]
Thanks，narusFR
 It fixed my issue.

**Comment by eff-society** [comment]
If Anaconda works normally after showing this error, just add the following to your Anaconda user settings:
**"swallow_startup_errors": true,**
